library/list.lisp

```diff
@@ -151,6 +154,13 @@
     ;; like (fold (flip Cons) Nil xs)
     (%reverse xs Nil))
 
+  ;; This is only for internal usage
+  (declare %reverse! (List :a -> List :a))
+  (define (%reverse! xs)
+    "A mutating reverse operation"
```

Comment:
even though this is only for internal usage, i'd like to add to the docstring the spec's caveat that `nreverse` is not required to (and won't) mutate the list in such a way that `(eq list (nreverse list))`, so you have to use the result and not re-use the original.

i think i'd add another line:

```
After (%reverse! LST), LST must not be referenced; the original list, and all its sublists, should be treated as consumed by this operation. Callers should use the return value and only the return value.
```